lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -375,10 +375,10 @@ def normal_void(authorization, options)
         end
       end
 
-      def add_payment_source(xml, source, action = nil)
+      def add_payment_source(xml, source, action = nil, options)
```

Comment:
The behavior of a mandatory parameter after an optional parameter can get really weird; I think it's worth reversing these (i.e. `add_payment_source(xml, source, options, action = nil)`).

That will also allow you to ax the `options=options` calls at 271 and 249 to boot.